examples/data_preprocessing/mimic_dataset/prepare_mimic_cxr.ipynb

```diff
@@ -0,0 +1,1862 @@
+    "print(rules_size[np.where(rules_size > 1)[0]])\n",
+    "# rule defib appears for two different classes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Z Matrix\n",
+    "The Z matrix contains the information about the rules matched to the reports. Its shape is #instances x #rules. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 45,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['rule_matches_z.lib']"
+      ]
+     },
+     "execution_count": 45,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "def get_rule_matches_z (data: np.ndarray, num_rules: int) -> np.ndarray:\n",
+    "    \"\"\"\n",
+    "    Function calculates the z matrix (samples x rules)\n",
+    "    data: np.array (reports)\n",
+    "    output: sparse z matrix\n",
+    "    \"\"\"\n",
+    "    rule_matches_z = np.zeros([len(data), num_rules])\n",
+    "    for ind in range(len(data)):\n",
+    "        for rule, rule_id in rule2rule_id.items():\n",
+    "            if rule in (data[ind]):\n",
+    "                rule_matches_z[ind, rule_id] = 1\n",
+    "    return rule_matches_z\n",
+    "\n",
+    "rule_matches_z = get_rule_matches_z(input_list[:,4], (len(rule2rule_id)+1))\n",
+    "\n",
+    "dump(rule_matches_z, Z)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Image encoding \n",
+    "For the image encoding we try two different approaches:\n",
+    "1. extract features using a pretrained model (no labels)\n",
+    "2. finetune a pretrained model and then extract features (finetuning with weak chexpert labels)\n",
+    "\n",
+    "## First approach: Extracting features using a pretrained model (no labels)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For the implementation we use PyTorch and the pretrained ResNet50 from `torchvision` library. \n",
+    "The following class loads the data and transforms it in the way it is required for [ResNet50](https://arxiv.org/abs/1512.03385). It is written in the form such that it is compatible with `torch.utils.data.DataLoader`. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 47,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class mimicDataset(Dataset):\n",
+    "    \n",
+    "    def __init__(self, path, load_labels = False):\n",
+    "        'Initialization'\n",
+    "        self.path = path\n",
+    "        self.load_labels = load_labels\n",
+    "        \n",
+    "    def __len__(self):\n",
+    "        'Denotes the total number of samples'\n",
+    "        return len(self.path)\n",
+    "    \n",
+    "    def __getitem__(self, index):\n",
+    "        'Generates one sample of data'\n",
+    "        # Select sample\n",
+    "        image = Image.open(self.path[index,3].replace(\".dcm\", \".jpg\")).convert('RGB')\n",
```

Comment:
Choose if you will use single or double quotes and keep the same throughout the code.